soa/service/s3.cc

```diff
@@ -44,6 +48,377 @@ using namespace ML;
 using namespace Datacratic;
 
 
+namespace {
+
+/****************************************************************************/
+/* S3 GLOBALS                                                               */
+/****************************************************************************/
+
+struct S3Globals {
+    S3Globals()
+        : baseRetryDelay(3), numRetries(-1)
+    {
+        if (numRetries == -1) {
+            char * numRetriesEnv = getenv("S3_RETRIES");
+            if (numRetriesEnv) {
+                numRetries = atoi(numRetriesEnv);
+            }
+            else {
+                numRetries = 45;
+            }
+        }
+
+        loop.start();
+    }
+
+    HttpClient & getClient(const string & bucket,
```

Comment:
We should mention the lifecycle and threading requirements in the comment for this method.